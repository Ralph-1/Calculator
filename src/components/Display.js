import PropTypes from 'prop-types';

function Display({ result }) {
  return (
    <div className="display">
      <div>{result}</div>
    </div>
  )
}

Display.defaultProps = { 
  result: '0',
};

Display.propTypes = { 
  result: PropTypes.string 
}

export default Display;