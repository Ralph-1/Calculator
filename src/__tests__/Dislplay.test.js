 
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Display from '../components/Display';

describe('Home component test', () => {
  test('should renders welcome text', () => {
    render(<Display />);
    const text = screen.getByText('0');
    expect(text).toBeInTheDocument();
  });

  test('should render the expected output', () => {
    const component = renderer.create(
      <Display />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});