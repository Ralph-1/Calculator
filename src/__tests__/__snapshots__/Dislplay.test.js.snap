// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component test should render the expected output 1`] = `
<div
  className="display"
>
  <h1>
    0
  </h1>
</div>
`;
